### PR TITLE
Configure metadata concealment iptables rules in node startup.

### PR DESCRIPTION
**Pull request by ikehz**
**What this PR does / why we need it**: Configure iptables rule for metadata concealment at startup so the pod doesn't have to, to reduce memory consumption.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by ihmccreery** [comment]
Xref https://github.com/GoogleCloudPlatform/k8s-metadata-proxy/pull/6, https://github.com/GoogleCloudPlatform/k8s-metadata-proxy/pull/7

**Comment by bowei** [comment]
@ihmccreery is this important to go into the current release

File: cluster/gce/configure-vm.sh [hunk 1]
```diff
@@ -86,11 +86,9 @@ ensure-local-disks() {
 function config-ip-firewall {
   echo "Configuring IP firewall rules"
 
-  iptables -N KUBE-METADATA-SERVER
-  iptables -I FORWARD -p tcp -d 169.254.169.254 --dport 80 -j KUBE-METADATA-SERVER
-
   if [[ "${ENABLE_METADATA_CONCEALMENT:-}" == "true" ]]; then
-    iptables -A KUBE-METADATA-SERVER -j DROP
+    echo "Add rule for metadata concealment"
+    iptables -t nat -I PREROUTING -p tcp -d 169.254.169.254 --dport 80 -j DNAT --to-destination 127.0.0.1:988
```
Comment (mikedanese) [inline cluster/gce/configure-vm.sh]: if you are removing the chain with a descriptive name, you should add a comment to this rule.

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -77,6 +70,11 @@ function config-ip-firewall {
     iptables -w -t nat -A IP-MASQ -d 192.168.0.0/16 -m comment --comment "ip-masq: local traffic is not subject to MASQUERADE" -j RETURN
     iptables -w -t nat -A IP-MASQ -m comment --comment "ip-masq: outbound traffic is subject to MASQUERADE (must be last in chain)" -j MASQUERADE
   fi
+
+  if [[ "${ENABLE_METADATA_CONCEALMENT:-}" == "true" ]]; then
+    echo "Add rule for metadata concealment"
+    iptables -t nat -I PREROUTING -p tcp -d 169.254.169.254 --dport 80 -j DNAT --to-destination 127.0.0.1:988
```
Comment (mikedanese) [inline cluster/gce/gci/configure-helper.sh]: use iptables -w and add a comment

File: cluster/gce/configure-vm.sh (hunk 1 above)
Comment (ihmccreery) [reply cluster/gce/configure-vm.sh]: Done.

File: cluster/gce/gci/configure-helper.sh (hunk 2 above)
Comment (ihmccreery) [reply cluster/gce/gci/configure-helper.sh]: Done.

**Comment by ihmccreery** [comment]
@bowei Yes, I'd like to see this get in for 1.9.

/retest

**Comment by ihmccreery** [comment]
@bowei @mikedanese Can we please get this into v1.9?

**Comment by ihmccreery** [comment]
Tests were failing because I hadn't shipped the v0.1.5 container yet.  Doh.

**Comment by bowei** [comment]
ike -- you need to add the kind and sig labels ideally

**Comment by jberkus** [comment]
@ihmccreery please add kind and sig labels so that we can include this in 1.9.  Thanks.

**Pull request merged**